src/org/zaproxy/zap/extension/ascan/ActiveScanAPI.java

```diff
@@ -366,6 +372,8 @@ public ApiResponse handleApiAction(String name, JSONObject params) throws ApiExc
 				}
 				policy = controller.getPolicyManager().getTemplatePolicy();
 				policy.setName(newPolicyName);
+				policy.setDefaultThreshold(getAlertThreshold(params));
```

Comment:
The alert threshold and attack strength should be set only when those are specified, otherwise we have to maintain the defaults in two places (or expose the defaults, but no point re-setting them in any case).